spring-boot-project/spring-boot/src/main/java/org/springframework/boot/logging/log4j2/Log4J2LoggingSystem.java

```diff
@@ -182,11 +182,12 @@ protected void loadConfiguration(String location, LogFile logFile) {
 	}
 
 	private ConfigurationSource getConfigurationSource(URL url) throws IOException {
-		InputStream stream = url.openStream();
-		if (FILE_PROTOCOL.equals(url.getProtocol())) {
-			return new ConfigurationSource(stream, ResourceUtils.getFile(url));
+		try (InputStream stream = url.openStream()) {
```

Comment:
This will close the `InputStream` before it can be retrieved from the `ConfigurationSource` and used. Log4j2 makes it the responsibility of the user of the `ConfigurationSource` to retrieve the `InputStream` and close it so this change is unnecessary.